src/ssl.c

```diff
@@ -15902,18 +15902,48 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
     {
         WOLFSSL_ENTER("wolfSSL_CTX_set_client_CA_list");
     #if defined(OPENSSL_EXTRA) || defined(WOLFSSL_EXTRA)
-        if (ctx != NULL)
+        if (ctx != NULL) {
+            wolfSSL_sk_X509_NAME_pop_free(ctx->ca_names, NULL);
             ctx->ca_names = names;
+        }
     #else
         (void)ctx;
         (void)names;
     #endif
     }
 
+    void wolfSSL_set_client_CA_list(WOLFSSL* ssl,
+                                       WOLF_STACK_OF(WOLFSSL_X509_NAME)* names)
+    {
+        WOLFSSL_ENTER("wolfSSL_set_client_CA_list");
+    #if defined(OPENSSL_EXTRA) || defined(WOLFSSL_EXTRA)
```

Comment:
Why is this same macro guard not needed for `wolfSSL_CTX_get_client_CA_list`?

Reply:
Not needed indeed. The functions are now protected by this guard. Removing redundant.